fenix/app/src/androidTest/java/org/mozilla/fenix/ui/DownloadTest.kt

```diff
@@ -238,4 +244,337 @@ class DownloadTest {
             assertExternalAppOpens("com.google.android.apps.docs")
         }
     }
+
+    // Download pdf file using the download toolbar button
+    @Test
+    fun downloadAndOpenPdfTest() {
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(downloadTestPage.toUri()) {
+            waitForPageToLoad()
+        }.clickDownloadLink(pdfFileName) {
+        }
+        browserScreen {
+        }.clickDownloadPDFButton {
+            verifyDownloadedFileName(pdfFileName)
+        }.clickOpen("application/pdf") {
+            assertExternalAppOpens("com.google.android.apps.docs")
```

Comment:
This is a hard reliance on `com.google.android.apps.docs` (Google Docs), can we set a constant in https://github.com/mozilla-mobile/firefox-android/blob/main/fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/Constants.kt and add a comment above the test that Google Docs is required.

@AndiAJ @sv-ohorvath Do you remember if we can assert before the test run that something is installed?